### PR TITLE
feat: add instant meeting parameter

### LINKED ISSUE #17660: [CAL-4734] Show instant meeting modal when params are set
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4734](https://linear.app/calcom/issue/CAL-4734/show-instant-meeting-modal-when-params-are-set)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes https://github.com/calcom/cal.com/issues/17660 (GitHub issue number)
- Fixes CAL-4734 (Linear issue number - should be visible at the bottom of the GitHub issue description)

Screenshot 2024-11-15 at 6 08 16 PM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1) Create a team org event types
2) Enabled instant bookings and add a parameter value and click save
3) Go to booking page and you should only see the modal when all parameter values are present

### CODE REVIEW

File: packages/features/bookings/Booker/Booker.tsx [hunk 1]
```diff
@@ -68,6 +69,12 @@ const BookerComponent = ({
   customClassNames,
 }: BookerProps & WrappedBookerProps) => {
   const { t } = useLocale();
+  const searchParams = useSearchParams();
+
+  const areInstantMeetingParametersSet = !!event.data?.instantMeetingParameters
+    ? event.data.instantMeetingParameters.every((param) => Array.from(searchParams.values()).includes(param))
+    : true;
+
```
Comment (Udit-takkar) [inline packages/features/bookings/Booker/Booker.tsx:77]: if there are instantMeetingParameters then all of them should be present

File: packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx [hunk 2]
```diff
@@ -171,6 +176,44 @@ export default function InstantEventController({
                             );
                           }}
                         />
+                        <div>
+                          <Label>{t("only_show_if_parameter_set")}</Label>
+                          <div className="space-y-2">
```
Comment (Udit-takkar) [inline packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:181]: In the next iteration i would be refactoring it to something like this

Screenshot 2024-11-15 at 5 47 26 PM

File: packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx [hunk 3]
```diff
@@ -76,6 +76,10 @@ export default function InstantEventController({
   const [instantEventState, setInstantEventState] = useState<boolean>(eventType?.isInstantEvent ?? false);
   const formMethods = useFormContext<FormValues>();
 
+  const [parameters, setParameters] = useState<string[]>(
+    formMethods.getValues("instantMeetingParameters") || []
+  );
+
```
Comment (Udit-takkar) [inline packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:82]: I tried using `useFieldArray` but it had some issues and type errors

File: packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx (hunk 2 above)
Comment (alishaz-polymath) [reply packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:180]: https://github.com/calcom/cal.com/pull/17361 Something similar is achieved here.

File: packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx (hunk 3 above)
Comment (alishaz-polymath) [reply packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:82]: What were the issues?

Comment (Udit-takkar) [reply packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:82]: - It was splitting the string to object like { 1: "I", 2: "N" , 3 :"D" }. 

and it's mentioned in the docs that useFieldArray doesn't support flat field arrays.

Comment (alishaz-polymath) [reply packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:82]: 🫠

**Review by alishaz-polymath — APPROVED** [review]
LGTM 🚀

**Pull request merged**